app/controllers/comments_controller.rb

```diff
@@ -1,5 +1,26 @@
 class CommentsController < ApplicationController
-  def index; end
+  def new
+    @comment = Comment.new
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+  end
 
-  def show; end
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.author_id = current_user.id
+    @comment.post_id = params[:post_id]
+
+    if @comment.save
+      redirect_to user_post_path(user_id: @comment.author_id, id: @comment.post_id),
```

Comment:
- [ ] You have a typo that is bringing an error message when I try to create a comment. Instead of `author_id`, it should be `user_id` .